x-pack/solutions/security/plugins/security_solution/public/flyout/document_details/shared/components/misconfiguration_insight.tsx

```diff
@@ -65,19 +65,19 @@ export const MisconfigurationsInsight: React.FC<MisconfigurationsInsightProps> =
     pageSize: 1,
   });
 
-  useEffect(() => {
-    if (telemetryKey) {
-      uiMetricService.trackUiMetric(METRIC_TYPE.COUNT, telemetryKey);
-    }
-  }, [telemetryKey, renderingId]);
-
   const passedFindings = data?.count.passed || 0;
   const failedFindings = data?.count.failed || 0;
   const totalFindings = useMemo(
     () => passedFindings + failedFindings,
     [passedFindings, failedFindings]
   );
-  const hasMisconfigurationFindings = totalFindings > 0;
+  const shouldRender = totalFindings > 0; // this component only renders if there are findings
+
+  useEffect(() => {
+    if (shouldRender && telemetryKey) {
```

Comment:
Have you considered typing `telemetryKey` it as required? Looks like it is always passed

Reply:
yeah makes sense, can do that